### ISSUE TITLE
android: add accessibility for visually impaired, screen reader

### ISSUE THREAD

**Issue by azurejoga** [libretro/RetroArch#16610]
## Description

Retroarch is not accessible to the visually impaired and is not accessible with a screen reader. When opening the retroarch application on Android, nothing happens. In other words, there is no output to the screen reader.
There is an accessibility option in Android retroarch, but nothing happens when activated!

### Expected behavior

When opening the application, all options should be read by the screen reader, and can be interacted with full

### Actual behavior

No options are read by the screen reader, literally what is read is gameview. that is, it is inaccessible

### Steps to reproduce the bug

1. Enable Google Talk Back screen reader.
2. open retroarch.
3. when you open it you will notice that nothing will be said, which it should be the other way around.

### Bisect Results

from the initial versions of retroarch

### Version/Commit
I have no idea since it's inaccessible
### Environment information

- OS: [android, 5 and 13.]
- Compiler: [none]
# other informations!
Other blind friends will comment on this issue helping and collaborating with more bugs and what we can do to help

**Comment by thgcode** [comment]
Hello,
I am not a developer of the project but here is what I have found so far. Adding Android accessibility (at least for the keyboard) might be possible by implementing the `accessibility_speak` and `is_narrator_running` callbacks on the `frontend/drivers/platform_unix.c` driver via JNI, by communicating with the android.speech.tts.TextToSpeech Java class. For the touch screen I have encountered the same results of not reading the menu options on IOS. I need to study the code more, but the touch screen interaction might be more difficult because Retroarch would need to create fake AccessibilityNodeInfo objects and override the "GameView" to expose them and do the same thing for IOS, or implement a screen reader inside Retroarch itself, this can solve the problem both for IOS and Android but would be a lot of work to implement.

**Comment by azurejoga** [comment]
@maddox  @justinweiss  @diegoviola 
These are good tips we can follow to get started, but I really wonder how much developers are actually interested in this...

**Comment by thgcode** [comment]
Hello,

I have implemented the first solution via the keyboard at the pull request #16638 . Currently the pull request is experimental and in progress because I am working on Talkback detection, it currently does not enable automatically like on IOS.

Currently if someone wants to test the change, you can download the apk on the PR and modify `accessibility_enabled = false` to true on the configuration file or enable it through the interface and in theory Talkback should read, at least via keyboard.

**Comment by azurejoga** [comment]
@thgcode  This is amazing, I'm going to test this now and I'll bring the feedback here!

**Comment by thgcode** [comment]
After some tests I managed to fix the Talkback detection. For the speak method I used the `announceForAccessibility` method that communicates directly with the screen reader, thus not requiring to implement TTS manually.

**Comment by azurejoga** [comment]
In my opinion, after testing, it is available to merge into the base branch
